ui/src/testing/stubs/profile-service-stub.ts

```diff
@@ -26,6 +31,12 @@ export class ProfileStubVariables {
   };
 }
 
+export class ProfileApiStub extends ProfileApi {
```

Comment:
More naming nits:

When reading through the test code that uses this class, I found it pretty confusing that a file named profile-service-stub exports a class named ProfileApiStub. It feels like this class is somewhat shoved in here, especially since there *is* an exported class that matches the filename.

Will some of this go away once Angular is gone?

Or maybe making the filename more generic would help, e.g. "testing/stubs/profile.ts" (since "stub" is implied by the path, and this file now exports both an Api stub and a Service stub for the profile API).